### ISSUE TITLE
Server Crash - Random Timing

### ISSUE THREAD

**Issue by chakri1vr1** [mcollina/mosca#320]
The Server code - running on a raspberry pi is crashing some times without any proper identifiable reason. The crash log is as below:

```javascript

{"name":"mosca","hostname":"raspberrypi","pid":7485,"client":"9_6","level":40,"err":{"message":"cannot parse topic","name":"Error","stack":"Error: cannot parse topic\n    at Parser._parsePublish (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/mqtt-packet/parser.js:247:31)\n    at Parser._parsePayload (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/mqtt-packet/parser.js:116:14)\n    at Parser.parse (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/mqtt-packet/parser.js:42:48)\n    at DestroyableTransform.process [as _transform] (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/lib/parseStream.js:13:12)\n    at DestroyableTransform.Transform._read (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:184:10)\n    at DestroyableTransform.Transform._write (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:172:12)\n    at doWrite (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:237:10)\n    at writeOrBuffer (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:227:5)\n    at DestroyableTransform.Writable.write (/home/pi/Desktop/server/node_modules/mosca/node_modules/mqtt-connection/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:194:11)\n    at Socket.ondata (_stream_readable.js:540:20)"},"msg":"cannot parse topic","time":"2015-07-25T06:24:02.342Z","v":0}
/home/pi/Desktop/server/node_modules/mosca/node_modules/qlobber/lib/qlobber.js:325
    return this._match([], 0, topic.split(this._separator), this._trie);
                                   ^
TypeError: Cannot read property 'split' of null
    at Qlobber.match (/home/pi/Desktop/server/node_modules/mosca/node_modules/qlobber/lib/qlobber.js:325:36)
    at EventEmitter.TrieAscoltatore.publish (/home/pi/Desktop/server/node_modules/mosca/node_modules/ascoltatori/lib/trie_ascoltatore.js:51:27)
    at EventEmitter.newPublish (/home/pi/Desktop/server/node_modules/mosca/node_modules/ascoltatori/lib/abstract_ascoltatore.js:144:20)
    at /home/pi/Desktop/server/node_modules/mosca/lib/server.js:321:22
    at EventEmitter.Server.storePacket (/home/pi/Desktop/server/node_modules/mosca/lib/server.js:422:5)
    at EventEmitter.publish (/home/pi/Desktop/server/node_modules/mosca/lib/server.js:315:8)
    at Client.handleAuthorizePublish (/home/pi/Desktop/server/node_modules/mosca/lib/client.js:487:15)
    at /home/pi/Desktop/server/node_modules/mosca/lib/client.js:104:14
    at EventEmitter.Server.authorizePublish (/home/pi/Desktop/server/node_modules/mosca/lib/server.js:378:3)
    at Connection.<anonymous> (/home/pi/Desktop/server/node_modules/mosca/lib/client.js:103:19)

```
It looks like the internal library which is trying to authenticate each packet has some problem or something similar. Can you please help me with debugging the issue ?

**Comment by mcollina** [comment]
Can you please upload some code that trigger this issue?

**Comment by chakri1vr1** [comment]
I can give you my code, but I feel it'd be very lengthy and useless. As you can see the Stack Trace refers only to the node_modules folders and nowhere to my actual code. That is what is weird. Let me know if giving the code can help you in any way.

**Comment by mcollina** [comment]
Probably, but this issue does not happen to me, so there is something particular that you are doing that trigger it. Without a way to reproduce, I cannot fix it.

**Comment by mcollina** [comment]
Which version of Mosca are you running? Have you updated to the latest one?

**Comment by eladnava** [comment]
@mcollina getting the same crash here, it happens about twice a day with 10k clients:

	/var/app/current/node_modules/mosca/node_modules/qlobber/lib/qlobber.js:325
		return this._match([], 0, topic.split(this._separator), this._trie);
									   ^
	TypeError: Cannot read property 'split' of undefined
		at Qlobber.match (/var/app/current/node_modules/mosca/node_modules/qlobber/lib/qlobber.js:325:36)
		at EventEmitter.TrieAscoltatore.publish (/var/app/current/node_modules/mosca/node_modules/ascoltatori/lib/trie_ascoltatore.js:51:27)
		at EventEmitter.newPublish (/var/app/current/node_modules/mosca/node_modules/ascoltatori/lib/abstract_ascoltatore.js:144:20)
		at /var/app/current/node_modules/mosca/lib/server.js:321:22
		at EventEmitter.Server.storePacket (/var/app/current/node_modules/mosca/lib/server.js:422:5)
		at EventEmitter.publish (/var/app/current/node_modules/mosca/lib/server.js:315:8)
		at Client.handleAuthorizePublish (/var/app/current/node_modules/mosca/lib/client.js:489:15)
		at /var/app/current/node_modules/mosca/lib/client.js:515:14
		at EventEmitter.Server.authorizePublish (/var/app/current/node_modules/mosca/lib/server.js:378:3)
		at Immediate._onImmediate (/var/app/current/node_modules/mosca/lib/client.js:514:19)

**Comment by eladnava** [comment]
@mcollina any chance this has been fixed in a newer mosca version? (package.json with `"mosca": "^0.30.0"`)

**Comment by eladnava** [comment]
@mcollina @chakri1vr1 I think the crash is fixed in `0.32.1`. I've yet to encounter it since updating from `0.30.0`.

**Comment by mcollina** [comment]
thanks for checking @eladnava
